software_bundle/software_bundle.proto

```diff
@@ -0,0 +1,193 @@
+syntax = "proto3";
+
+package gnoi.software_bundle;
+
+import "github.com/openconfig/gnoi/types/types.proto";
+
+option go_package = "github.com/openconfig/gnoi/software_bundle";
+
+option (types.gnoi_version) = "0.1.0";
+
+// SoftwareBundle is a service for managing software bundles on the system.
+// A "software bundle" is a collection of files (e.g. RPMs, squashfs)
+// which are installed to modify the behaviour of the processes on the
+// system.
+// Only one RPC (ListSoftwareBundles, FinalizeSoftwareBundles,
+// InstallSoftwareBundle, UninstallSoftwareBundle)
+// may be in progress at any given time
+service SoftwareBundle {
+
+  // Shows the current state of the software bundles on the system.
+  rpc ListSoftwareBundles(ListSoftwareBundlesRequest)
+      returns (ListSoftwareBundlesResponse);
+
+  // FinalizeSoftwareBundles reloads any processes on the system for
+  // which software bundles have been installed but which have not yet been
+  // reloaded with those software bundles.
+  // Please note that this may kill the connection if the process
+  // hosting the gRPC server reloads, so the server is not guaranteed
+  // to return a response here.
+  rpc FinalizeSoftwareBundles(FinalizeSoftwareBundlesRequest)
```

Comment:
Minor suggestion:

```suggestion
  rpc ActivateSoftwareBundles(FinalizeSoftwareBundlesRequest)
```

Reply:
this RPC was renamed to `Activate` in the latest patch set